service/migration_manager.cc

```diff
@@ -103,19 +103,20 @@ void migration_manager::init_messaging_service()
             return db::schema_tables::recalculate_schema_version(_sys_ks, _storage_proxy.container(), _feat);
         });
     };
+    auto reload_schema_in_bg = [this] {
+        (void) with_gate(_background_tasks, [this] {
+            return reload_schema();
+        });
+    };
```

Comment:
> to eliminate our dependence on (unguaranteed) feature order

The fact that feature order is unguaranteed is worrisome, and should probably be fixed. In pariciular, this `digest_insensitive_to_expiry` feature is old and is supposed to be always enabled in new clusters. When introducing a new feature, we may reasonably assume that the cluster will have the older features enabled too. We make our lives harder by not guaranteeing this.

Reply:
That was my question as well - at which point can we stop negotiating a feature and assume it is unconditionally enabled on all nodes?

Reply:
> That was my question as well - at which point can we stop negotiating a feature and assume it is unconditionally enabled on all nodes?

We got rid of some always-enabled features when all supported releases enable the feature.